include/gdf/cffi/functions.h

```diff
@@ -34,544 +34,8 @@ int64_t gdf_ipc_parser_get_data_offset(gdf_ipc_parser_type *handle);
-size_t gdf_join_result_size(gdf_join_result_type *result);
-
-/* prefixsum */
-
-gdf_error gdf_prefixsum_generic(gdf_column *inp, gdf_column *out, int inclusive);
-gdf_error gdf_prefixsum_i8(gdf_column *inp, gdf_column *out, int inclusive);
-gdf_error gdf_prefixsum_i32(gdf_column *inp, gdf_column *out, int inclusive);
-gdf_error gdf_prefixsum_i64(gdf_column *inp, gdf_column *out, int inclusive);
-
-
-/* unary operators */
-
-/* hashing */
-
-// num_cols: the number of columns
-// input: a list of the column pointers
-// hash: the hashing function to use
-// output: the output column, allocated by the caller, must have GDF_INT32 dtype
-gdf_error gdf_hash(int num_cols, gdf_column **input, gdf_hash_func hash, gdf_column *output);
-
-/* trig */
-
-gdf_error gdf_sin_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_sin_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_sin_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_cos_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_cos_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_cos_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_tan_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_tan_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_tan_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_asin_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_asin_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_asin_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_acos_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_acos_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_acos_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_atan_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_atan_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_atan_f64(gdf_column *input, gdf_column *output);
-
-/* exponential */
-
-gdf_error gdf_exp_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_exp_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_exp_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_log_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_log_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_log_f64(gdf_column *input, gdf_column *output);
-
-/* power */
-
-gdf_error gdf_sqrt_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_sqrt_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_sqrt_f64(gdf_column *input, gdf_column *output);
-
-
-/* rounding */
-
-gdf_error gdf_ceil_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_ceil_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_ceil_f64(gdf_column *input, gdf_column *output);
-
-gdf_error gdf_floor_generic(gdf_column *input, gdf_column *output);
-gdf_error gdf_floor_f32(gdf_column *input, gdf_column *output);
-gdf_error gdf_floor_f64(gdf_column *input, gdf_column *output);
-
-/* casting */
```

Comment:
I can see casting going either way. It sort of is a compute in that it operates over a column. On the other hand it does seem fairly fundemental. I'm happy with removing them for now, and adding back as needed.